exercise/templatetags/exercise.py

```diff
@@ -117,81 +117,68 @@ def percent(decimal):
 def submission_status(status):
     return Submission.STATUS[status]
 
-
+def _get_points_type(context):
+    points_type = ""
+
+    if ((context['confirm_the_level']) or
+        context['difficulty'] == "-" and context['max'] <= 2):
+
+        if (context['full_score'] or
+            (context['passed'] and context['confirm_the_level'])):
```

Comment:
I need to take a paper and draw this logic paths on it. :D

Reply:
I guess it is a little messy but I did not feel like making a huge bunch of `if...elif...elif...else` -type structures where some of the blocks could have the same contents. Also, I tried to copy the logic from the `points_badge` template as accurately as possible but that was a bit messy too so this might be more complicated than it would need to be.

Reply:
Set would work here bettere, or just two variables. Sets are nice as you can then do:

`s.add("plaah")` and `"plaah" in s` and it's way faster than with lists.

Reply:
I do see your point - on the other hand, the set will contain 1 or 2 elements: some string and, in some cases, "unconfirmed". I can easily check if "unconfirmed" exists in the set with the in-keyword but the other string is one of numerous alternatives and it doesn't seem very appealing to go over all possible alternatives to see if they are contained. Also, if the list has length 1 or 2, does it actually create a big performance problem?

Reply:
I think I got it to work using a set.